src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php

```diff
@@ -145,8 +147,13 @@ public static function castFrameStub(FrameStub $frame, array $a, Stub $stub, $is
                 $f['file'] = substr($f['file'], 0, -strlen($match[0]));
                 $f['line'] = (int) $match[1];
             }
+            $relFile = $f['file'];
+            $home = getenv('HOME');
+            if (false !== $home && 0 === strpos($relFile, $home.DIRECTORY_SEPARATOR)) {
+                $relFile = '~'.DIRECTORY_SEPARATOR.substr($relFile, strlen($home) + 1);
```

Comment:
can be simply ``'~'.substr($relFile, strlen($home))``. No need to remove the directory separator to add it again